### PR TITLE
Create coverage tests for `array` package

### PR DESCRIPTION
**Pull request by vinhpad**
# Summary
This PR adds unit tests to improve the test coverage for the following functions in the `array` package:

- `Map(array []T, mapper func(T) R) []R`
- `Contains(array []T, element T) bool`
- `Find(array []T, predicate func(T) bool) (T, bool)`

# Details

## Map tests
- Verifies correct mapping of array elements with different types:
  - Integer array transformation (multiplying by 2)
  - String array to integer array conversion

## Contains tests
- Tests element existence checking with different data types:
  - Integer arrays
  - String arrays
  - Interface arrays (any type)
- Handles both positive (element exists) and negative (element not found) cases

## Find tests
- Tests element search with predicate functions:
  - Integer arrays with number comparison
  - String arrays with exact matching
  - Interface arrays with type assertion
- Handles both success and failure cases appropriately

# Test Coverage
- All logical branches of the three functions are now covered
- Both positive and negative test cases are included
- Edge cases are properly handled

### CODE REVIEW

File: utils/array/array_test.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		array := []string{"a", "b", "c"}
+		predicate := func(s string) bool {
+			return s == "d"
+		}
+		got, found := Find(array, predicate)
+		assert.False(t, found)
+		assert.Equal(t, "", got)
+	})
+
+	t.Run("Find any", func(t *testing.T) {
```
Comment (seungduk-yanolja) [inline utils/array/array_test.go:70]:
```
Find a value from a mixed-type array
```
or something else because `find any` sounds like `find anything comes first`

**Review by seungduk-yanolja — APPROVED** [review]

File: utils/array/array_test.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	t.Run("Find a value from a mixed-type array", func(t *testing.T) {
+		array := []any{1, "test", 3.14, []int{1, 2, 3}}
+		predicate := func(a any) bool {
+			return a == "test"
+		}
+		got, found := Find(array, predicate)
+		assert.True(t, found)
+		assert.Equal(t, "test", got)
+	})
+
```
Comment (seungduk-yanolja) [inline utils/array/array_test.go:79]: remove this line please
